Gems/Prefab/PrefabDependencyViewer/Code/Source/MainWindow.cpp

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Interface/Interface.h>
+#include <Core/Core.h>
+#include <MainWindow.h>
+#include <QBoxLayout>
+#include <QLabel>
+#include <GraphCanvas/GraphCanvasBus.h>
+#include <GraphCanvas/Components/Nodes/NodeTitleBus.h>
+#include <GraphCanvas/Components/Nodes/NodeBus.h>
+
+namespace PrefabDependencyViewer
+{
+    GraphCanvas::GraphCanvasTreeItem* PrefabDependencyViewerConfig::CreateNodePaletteRoot()
+    {
+        const GraphCanvas::EditorId& editorId = PREFAB_DEPENDENCY_VIEWER_EDITOR_ID;
+        GraphCanvas::NodePaletteTreeItem* rootItem = aznew GraphCanvas::NodePaletteTreeItem("Root", editorId);
+
+        return rootItem;
+    }
+
+    /** Returns a bare minimum configuration to configure GraphCanvas UI
+    for the purpose of visualizing the Prefab hierarchy. */
+    PrefabDependencyViewerConfig* GetDefaultConfig()
+    {
+        // Make sure no memory leak happens here
+        PrefabDependencyViewerConfig* config = new PrefabDependencyViewerConfig();
+        config->m_editorId = PREFAB_DEPENDENCY_VIEWER_EDITOR_ID;
+        config->m_baseStyleSheet = "PrefabDependencyViewer/StyleSheet/graphcanvas_style.json";
+
+        return config;
+    }
+
+    PrefabDependencyViewerWidget::PrefabDependencyViewerWidget(QWidget* pParent)
+        : GraphCanvas::AssetEditorMainWindow(GetDefaultConfig(), pParent)
+    {
+        AZ::Interface<PrefabDependencyViewerInterface>::Register(this);
+    }
+
+    void PrefabDependencyViewerWidget::SetupUI() {
+        GraphCanvas::AssetEditorMainWindow::SetupUI();
+        delete m_nodePalette;
+        m_nodePalette = nullptr;
+    }
```

Comment:
May be worth adding a comment here explaining that we're using the basic Graph Canvas editor window but not using the ability to add nodes, so will be deleting that control.

(Do we also suppress the ability to rihgt click to add nodes, delete nodes, etc?)

Reply:
You can still right click on a node and delete that node but you can't add a node right now. The tool is still an experimental feature and needs customer validation so I don't know if we want to add the functionality of creating and deleting nodes from the UI before displaying assets in the hierarchy.

Reply:
Resolved